tools/OpenConsole.psm1

```diff
@@ -363,16 +363,91 @@ function Invoke-ClangFormat {
     }
 }
 
+function Invoke-StripBOM {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
+        [string]$Path
+    )
+
+    $files = Get-ChildItem -Path "$Path\*" -Include *.xaml
+    foreach ($file in $files) {
+        $content = Get-Content $file
+        [IO.File]::WriteAllLines($file.FullName, $content)
+    }
+}
+
+
 #.SYNOPSIS
-# runs code formatting on all c++ files
+# Check that xaml files are formatted correctly. This won't actually
+# format the files - it'll only ensure that they're formatted correctly.
+function Invoke-VerifyXamlFormat() {
+    $root = Find-OpenConsoleRoot
+    & dotnet tool restore --add-source https://api.nuget.org/v3/index.json
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalApp" --passive
+    if ($lastExitCode -eq 1) {
+        throw "Xaml formatting bad, run Invoke-XamlFormat on branch"
+    }
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalControl" --passive
+    if ($lastExitCode -eq 1) {
+        throw "Xaml formatting bad, run Invoke-XamlFormat on branch"
+    }
+
+    dotnet tool run xstyler -- -c "$root\Settings.XamlStyler" -d "$root\src\cascadia\TerminalSettingsEditor" --passive

```

Comment:
Does xstyler take multiple directory arguments, or does it just somewhat maddeningly only take a single directory?

If we ever add another folder of xaml files, this will not be excellent.

I tried to find all folders containing .xaml files and unfortunately hit the WPF control _and all of CascadiaPackage/'s build content_ ugh ugh ugh

Reply:
Oh, it's the maddening answer all right 😠. I know that this is pain, but like you've noticed, no good way of actually just getting the XAMLs we want.